pkg/internal/apis/certmanager/validation/certificate_test.go

```diff
@@ -509,14 +557,78 @@ func TestValidateCertificate(t *testing.T) {
 					RevisionHistoryLimit: int32Ptr(0),
 				},
 			},
+			a: someAdmissionRequest,
 			errs: []*field.Error{
 				field.Invalid(fldPath.Child("revisionHistoryLimit"), int32(0), "must not be less than 1"),
 			},
 		},
+		"v1alpha2 certificate created": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "abc",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+				},
+			},
+			a: &admissionv1.AdmissionRequest{
+				Kind: metav1.GroupVersionKind{Group: "cert-manager.io",
+					Version: "v1alpha2",
+					Kind:    "Certificate"},
+			},
+			warnings: validation.WarningList{
+				fmt.Sprintf(deprecationMessageTemplate,
+					cmapiv1alpha2.SchemeGroupVersion.String(),
+					"Certificate",
+					cmapi.SchemeGroupVersion.String(),
+					"Certificate"),
+			},
+		},
+		"v1alpha3 certificate created": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "abc",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+				},
+			},
+			a: &admissionv1.AdmissionRequest{
+				Kind: metav1.GroupVersionKind{Group: "cert-manager.io",
+					Version: "v1alpha3",
+					Kind:    "Certificate"},
+			},
+			warnings: validation.WarningList{
+				fmt.Sprintf(deprecationMessageTemplate,
+					cmapiv1alpha3.SchemeGroupVersion.String(),
+					"Certificate",
+					cmapi.SchemeGroupVersion.String(),
+					"Certificate"),
+			},
+		},
+		"v1beta1 certificate created": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "abc",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+				},
+			},
+			a: &admissionv1.AdmissionRequest{
+				Kind: metav1.GroupVersionKind{Group: "cert-manager.io",
+					Version: "v1alpha3",
+					Kind:    "Certificate"},
+			},
+			warnings: validation.WarningList{
+				fmt.Sprintf(deprecationMessageTemplate,
+					cmapiv1alpha3.SchemeGroupVersion.String(),
+					"Certificate",
+					cmapi.SchemeGroupVersion.String(),
+					"Certificate"),
+			},
+		},
```

Comment:
suggestion: is there a copy-paste mistake here? the test description is `v1beta1` but the test refers to `v1alpha3` and `cmapiv1alpha3`

Reply:
Thanks for spotting that! I've fixed it